### PR TITLE
docs(action-sheet): component playground examples

### PR DESCRIPTION
**Pull request by sean-perkins**
Introduces component playground examples for `ion-action-sheet`:
- Basic: Usage with many buttons, data and roles
- Theming > Styling: Example targeting specific elements to customize the appearance
- Theming > CSS Custom Properties: Example using CSS variables to customize the appearance

Quick preview link: https://ionic-docs-git-fw-1309-ionic1.vercel.app/docs/api/action-sheet

### CODE REVIEW

File: static/usage/action-sheet/theming/styling/demo.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  <script type="module" src="https://cdn.jsdelivr.net/npm/@ionic/core@6/dist/ionic/ionic.esm.js"></script>
+  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@ionic/core@6/css/ionic.bundle.css" />
+
+  <style>
+    ion-action-sheet.my-custom-class .action-sheet-group {
+      background: #F58840;
+    }
+
+    ion-action-sheet.my-custom-class .action-sheet-title {
+      color: #EADEDE;
```
Comment (liamdebeasi) [inline static/usage/action-sheet/theming/styling/demo.html:19]: Can we keep the title/subtitle as white? The gray is a bit hard to read on the orange background.

Also looks like some of the active indicators are not styled: 
image

Comment (liamdebeasi) [reply static/usage/action-sheet/theming/styling/demo.html:19]: Also why is the "Delete" button white on iOS and black on MD when the rest of the buttons are always black?

File: static/usage/action-sheet/basic/vue.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  code {
+    white-space: pre-wrap;
+  }
+</style>
+
+<template>
+  <div class="container">
+    <ion-button @click="presentActionSheet">Open</ion-button>
+    <code>{{result}}</code>
```
Comment (liamdebeasi) [inline static/usage/action-sheet/basic/vue.md:19]:
```suggestion
    <code>{{ result }}</code>
```

File: static/usage/action-sheet/basic/vue.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  import { IonButton, actionSheetController } from '@ionic/vue';
+
+  export default {
+    components: { IonButton },
+    data() {
+      return {
+        result: '',
+      };
+    },
+    methods: {
```
Comment (liamdebeasi) [inline static/usage/action-sheet/basic/vue.md:33]: Any particular reason why the Options API was chosen over the Composition API?

File: static/usage/action-sheet/theming/css-properties/demo.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      },
+      {
+        text: 'Cancel',
+        role: 'cancel',
+        data: {
+          action: 'cancel'
+        }
+      }];
+
+      document.body.appendChild(actionSheet);
```
Comment (liamdebeasi) [inline static/usage/action-sheet/theming/css-properties/demo.html:60]: This should be attached to `ion-app`. Alternatively, you can use the alertController which will do this for you. (same goes for the other instances of `document.body.appendChild`)

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

File: static/usage/action-sheet/theming/css-properties/demo.html (hunk 4 above)
Comment (liamdebeasi) [reply static/usage/action-sheet/theming/css-properties/demo.html:60]: Hmm looks like alert also does `document.body`: https://ionicframework.com/docs/api/alert

I'd be fine doing `document.body` for now to be consistent and then revisiting the best practice later.

File: static/usage/action-sheet/basic/vue.md (hunk 3 above)
Comment (sean-perkins) [reply static/usage/action-sheet/basic/vue.md:33]: I thought we were waiting until all component playgrounds exist, before moving to the Composition API usage?

File: static/usage/action-sheet/theming/css-properties/demo.html (hunk 4 above)
Comment (sean-perkins) [reply static/usage/action-sheet/theming/css-properties/demo.html:60]: Honestly - I referenced `ion-alert` examples when creating these examples. I can update both to match a specific usage pattern, if we have one.

File: static/usage/action-sheet/basic/vue.md (hunk 3 above)
Comment (liamdebeasi) [reply static/usage/action-sheet/basic/vue.md:33]: I think that was for using `script setup`, not the Composition API. Our current usage looks to be inconsistent:

Composition API: https://ionicframework.com/docs/api/accordion#toggle-accordions
Options API: https://ionicframework.com/docs/api/alert#controller

Comment (sean-perkins) [reply static/usage/action-sheet/basic/vue.md:33]: Ahh good distinction - I'll get that fixed here + get `ion-alert` updated as well.

File: static/usage/action-sheet/theming/styling/demo.html (hunk 1 above)
Comment (sean-perkins) [reply static/usage/action-sheet/theming/styling/demo.html]: Updated based on feedback.

The "Delete" button was white to show how to override the color opinionation we make for iOS mode. I removed this from the example, as the styles do not apply on MD & we do not currently support a CSS variable for developers to easily customize that appearance.

File: docs/api/action-sheet.md [hunk 5]
```diff
@@ -1,12 +1,8 @@
 ---
 title: "ion-action-sheet"
-hide_table_of_contents: true
-demoUrl: "/docs/demos/api/action-sheet/index.html"
-demoSourceUrl: "https://github.com/ionic-team/ionic-docs/tree/main/static/demos/api/action-sheet/index.html"
 ---
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
```
Comment (amandaejohnston) [inline docs/api/action-sheet.md:5]: `Tabs` and `TabItem` aren't used anymore here, so we can remove these imports.

File: docs/api/action-sheet.md [hunk 6]
```diff
@@ -24,28 +20,26 @@ import EncapsulationPill from '@components/page/api/EncapsulationPill';
...
-  toc={toc}
-  maxHeadingLevel={2}
-/>
 
+An Action Sheet is a dialog that displays a set of options. It appears on top of the app's content, and must be manually dismissed by the user before they can resume interaction with the app. Destructive options are made obvious in `ios` mode. There are multiple ways to dismiss the action sheet, including tapping the backdrop or hitting the escape key on desktop.
 
+import Basic from '@site/static/usage/action-sheet/basic/index.md';
 
-An Action Sheet is a dialog that displays a set of options. It appears on top of the app's content, and must be manually dismissed by the user before they can resume interaction with the app. Destructive options are made obvious in `ios` mode. There are multiple ways to dismiss the action sheet, including tapping the backdrop or hitting the escape key on desktop.
+<Basic />
```
Comment (amandaejohnston) [inline docs/api/action-sheet.md:28]: The result display doesn't seem to be working in the Vue Stackblitz 🤔

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

**Review by liamdebeasi — COMMENTED** [review]
A few things I noticed:

1. In the "CSS Custom Properties" demo, the "Delete" button is red on iOS, but black on MD:

| ios | md |
| - | - |
| image | image |

Maybe all the button texts should be white?

2. In the "Styling" demo, the "Delete" button is white but all the other buttons are black. Maybe both the "Styling" and "CSS Custom Properties" demos should be aligned on button colors?

**Comment by sean-perkins** [comment]
@liamdebeasi 

1. That is the default experience for destructive role buttons. On iOS we render the destructive role as red to align with the iOS design spec. MD does not have that opinionation. We do not expose a CSS variable to customize this appearance (other than overriding the danger color, which is only applied in iOS).
2. I can adjust this one, as we can target the selectors easily.

**Review by amandaejohnston — APPROVED** [review]

**Comment by liamdebeasi** [comment]
> That is the default experience for destructive role buttons. On iOS we render the destructive role as red to align with the iOS design spec. MD does not have that opinionation. We do not expose a CSS variable to customize this appearance (other than overriding the danger color, which is only applied in iOS).

We already override the danger color in the "Styling" demo:

```css
ion-action-sheet.my-custom-class .action-sheet-button,
ion-action-sheet.my-custom-class .action-sheet-button.ion-focused {
  color: #000000;
}
```

image

The red button on the orange background is hard to read, so we should either make the button black or change the background color of the action sheet.

| red text | black text |
| - | - |
| Screen Shot 2022-09-26 at 9 53 48 AM | Screen Shot 2022-09-26 at 9 53 39 AM |

**Pull request merged**